src/core/io/outputfb.h

```diff
@@ -0,0 +1,183 @@
+        var_QI = paQI;
+        var_PARAMS = paPARAMS;
+        var_OUT = paOUT;
+        receiveInputEvent(scmEventREQID, nullptr);
+        paQO = var_QO;
+        paSTATUS = var_STATUS;
+      }
+
+    protected:
+      static const SFBInterfaceSpec scmFBInterfaceSpec;
+
+      void onHandle(IOHandle *const paHandle) final override {
+        CProcessInterfaceFB::onHandle(paHandle);
+        if(isReady()) {
+          var_QO = write();
+        }
+      }
+
+      CDataConnection **getDIConUnchecked(TPortId paIndex) override final {
+        if(paIndex == 2) {
+          return &conn_OUT;
+        }
+        return CProcessInterfaceFB::getDIConUnchecked(paIndex);
+      }
+
+    private:
+      static const CStringDictionary::TStringId scmDataInputNames[];
+      static const CStringDictionary::TStringId scmDataInputTypeIds[];
+      static const CStringDictionary::TStringId scmDataOutputNames[];
+      static const CStringDictionary::TStringId scmDataOutputTypeIds[];
+      static const TDataIOID scmEIWith[];
+      static const TForteInt16 scmEIWithIndexes[];
+      static const CStringDictionary::TStringId scmEventInputNames[];
+      static const CStringDictionary::TStringId scmEventInputTypeIds[];
+      static const TDataIOID scmEOWith[];
+      static const TForteInt16 scmEOWithIndexes[];
+      static const CStringDictionary::TStringId scmEventOutputNames[];
+      static const CStringDictionary::TStringId scmEventOutputTypeIds[];
+
+      void executeEvent(TEventID paEIID, CEventChainExecutionThread *const paECET) final override {
+        if(paEIID == scmEventREQID) {
+          if(var_QI) {
+            var_QO = write();
+          } else {
+            var_QO = false_BOOL;
+          }
+          sendOutputEvent(scmEventCNFID, paECET);
+        } else {
+          CProcessInterfaceFB::executeEvent(paEIID, paECET);
+        }
+      }
+
+      void readInputData(TEventID paEIID) final override {
+        switch(paEIID){
+          case scmEventREQID: {
+            readData(0, var_QI, conn_QI);
+            readData(2, var_OUT, conn_OUT);
+            break;
+          }
+          default:
+            CProcessInterfaceFB::readInputData(paEIID);
+            break;
+        }
+      }
+
+      void setInitialValues() final override {
+        CProcessInterfaceFB::setInitialValues();
+        var_OUT = T(0);
+      }
+
+      CIEC_BOOL write() {
+        auto curHandle = getHandle();
+        if(!isReady() || curHandle == nullptr) {
+          return false_BOOL;
+        }
+
+        curHandle->set(var_OUT);
+
+        return true_BOOL;
+      }
+
+  };
+
+  template<class T>
+  const CStringDictionary::TStringId COutputFB<T>::scmDataInputNames[] = {STRID(QI), STRID(PARAMS), STRID(OUT)};
+  template<class T>
+  const CStringDictionary::TStringId COutputFB<T>::scmDataInputTypeIds[] = {STRID(BOOL), STRID(STRING), STRID(BYTE)};
```

Comment:
I believe this should now be:
```suggestion
  const CStringDictionary::TStringId COutputFB<T>::scmDataInputTypeIds[] = {STRID(BOOL), STRID(STRING), forte::CDataTypeTrait<T>::scmDataTypeName};
```

Reply:
Thanks for noting my copy paste mistake. Fixed in the update.